src/gui/dialogs/achievements_dialog.cpp

```diff
@@ -0,0 +1,102 @@
+/*
+	Copyright (C) 2003 - 2022
+	by David White <dave@whitevine.net>
+	Part of the Battle for Wesnoth Project https://www.wesnoth.org/
+
+	This program is free software; you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation; either version 2 of the License, or
+	(at your option) any later version.
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY.
+
+	See the COPYING file for more details.
+*/
+
+#define GETTEXT_DOMAIN "wesnoth-lib"
+
+#include "gui/dialogs/achievements_dialog.hpp"
+
+#include "gettext.hpp"
+#include "gui/auxiliary/find_widget.hpp"
+#include "gui/widgets/window.hpp"
+#include "log.hpp"
+
+namespace gui2::dialogs
+{
+
+REGISTER_DIALOG(achievements_dialog)
+
+achievements_dialog::achievements_dialog()
+: modal_dialog(window_id())
+, achieve_()
+, achievements_box_(nullptr)
+, content_names_(nullptr)
+{
+
+}
+
+void achievements_dialog::pre_show(window& win)
+{
+	std::vector<config> content_list;
+	content_names_ = &find_widget<menu_button>(&win, "selected_achievements_list", false);
+	connect_signal_notify_modified(*content_names_, std::bind(&achievements_dialog::set_achievements_content, this));
+
+	achievements_box_ = find_widget<listbox>(&win, "achievements_list", false, true);
+
+	for(const auto& [content_for, list] : achieve_.get_list())
```

Comment:
As it's a new file, run it through clang-format unless it's so similar to another file that we should be doing a side-by-side comparison.

Reply:
Updated.